libnavigation-core/src/main/java/com/mapbox/navigation/core/lifecycle/MapboxNavigationApp.kt

```diff
@@ -51,6 +52,10 @@ import kotlin.reflect.KClass
  *   }
  * }
  * ```
+ *
+ * NOTE: the implementation is not thread-safe. Some of the methods are required to be called
+ * from the main thread. Others can be called from any thread, however, it's user's responsibility
+ * to guarantee thread safety.
```

Comment:
```suggestion
```

Adding `@UIThread` is good for instructions.

Reply:
But some methods are not annotated. For example, `isSetup` can be called from any thread, while `setup` - only from main thread. Shouldn't we specify that it is not thread-safe to invoke these methods from different threads? Another option is to add `@UiThread` annotation to `isSetup` (and other not thread-safe methods) as well.

Reply:
`getObserver` and `current` is thread safe for example.

`isSetup`.. That sounds like a fine reason to add `@UiThread` to the `isSetup` function.

Reply:
Or put the note on the `isSetup` function.

Reply:
Discussed internally.